### PR TITLE
Add DoubleCheck as an explicit variant

### PR DESCRIPTION
**Pull request by chris-wood**

### CODE REVIEW

File: draft-wood-key-consistency.md [hunk 1]
```diff
@@ -193,13 +194,49 @@ below.
...
+|          |
+|  Client  +-----------+
+|          |           |
++----------+           |
+                       v
++----------+         +-----------+       +----------+
+|          |         | Untrusted |       |          |
+|  Client  +-------->+   Proxy   +------>+  Server  |
+|          |         |           |       |          |
+|          ==============================>          |
```
Comment (martinthomson) [inline draft-wood-key-consistency.md:229]:
```suggestion
|          +============================>+          |
|          |         |           |       |          |
```

Sometimes you need to test aasvg to see what it produces.

**Review by martinthomson — COMMENTED** [review]
Do you want to talk about the reason that this is a connect tunnel using the same proxy?  I think that this is a central piece of Ben's draft.  The idea is that the proxy is integral to the design as that proxy is later used for protocol interactions that use the KCCS target.  Because use of the proxy is part of the overall configuration that can be used to distinguish users, this arrangement doesn't expand the threat profile for users.

**Comment by martinthomson** [comment]
Oh, I forgot.  I don't think that "untrusted" is a good label here.  This requires less trust than the other arrangement, but you still need to trust that the proxy doesn't collude with the server.

**Comment by chris-wood** [comment]
> Do you want to talk about the reason that this is a connect tunnel using the same proxy? I think that this is a central piece of Ben's draft. The idea is that the proxy is integral to the design as that proxy is later used for protocol interactions that use the KCCS target. Because use of the proxy is part of the overall configuration that can be used to distinguish users, this arrangement doesn't expand the threat profile for users.

I don't agree with the requirement to use the same proxy for the connect tunnel, so I left that part out. (See OHAI mailing list.)

> Oh, I forgot. I don't think that "untrusted" is a good label here. This requires less trust than the other arrangement, but you still need to trust that the proxy doesn't collude with the server.

Yeah, I couldn't think of a better label for it. Maybe we go with something like "Single Proxy with Key Confirmation"?

**Pull request merged**